lib/puppet/provider/service/launchd.rb

```diff
@@ -298,13 +317,21 @@ def enabled?
   # without actually loading/unloading the job.
   def enable
     overrides = self.class.read_plist(self.class.launchd_overrides)
-    overrides[resource[:name]] = { "Disabled" => false }
+    if self.class.get_os_version < 14
+      overrides[resource[:name]] = { "Disabled" => false }
+    else
+      overrides[resource[:name]] = false
+    end
     Plist::Emit.save_plist(overrides, self.class.launchd_overrides)
   end
 
   def disable
-    job_path, job_plist = plist_from_label(resource[:name])
-    job_plist["Disabled"] = true
-    Plist::Emit.save_plist(job_plist, job_path)
+    overrides = self.class.read_plist(self.class.launchd_overrides)
```

Comment:
@kylog this removal was your work in Puppet 4 cleanup, but it looks to me like it chose the wrong piece to remove. The `enable` method kept the OS X 10.6 code branch (`has_macosx_plist_overrides? == true`) while the `disable` method kept the older `has_macosx_plist_overrides? == false` branch.